monai/apps/utils.py

```diff
@@ -327,7 +327,17 @@ def download_and_extract(
             be False.
         progress: whether to display progress bar.
     """
+    urlFilenameExtension = ''.join(Path(".", _basename(url)).resolve().suffixes)
+    if filepath:
+        FilepathExtenstion = ''.join(Path(".", _basename(filepath)).resolve().suffixes)
+        if urlFilenameExtension != FilepathExtenstion:
+            raise NotImplementedError(
```

Comment:
In my opinion, the issue [here](https://github.com/Project-MONAI/MONAI/issues/5463) is that if filepath can directly get the name of the downloaded file, the `download_and_extract` function would work as expected, rather than only raising an error at this point.

Reply:
I know your point here.
Do you think when user specify `filepath='./test'`, it would be better to give a warning and automatically catch the filename?

cc @ericspod @Nic-Ma

Reply:
I would warn instead of forcing extensions, it takes control away from people to force things like this that aren't strictly necessary.

Reply:
But I have a question.
Is it necessary for the user to specify the file extension when setting the `filepath` for the downloaded file?

Reply:
If the user gives a path without extension then you'll get a file with no extension, the user would have to account for that and provide an extension. You could add an extension if none is given (and warn that this happened), but if the user has given a different extension than what's expected, this should be a warning and not an error. If you add an extension you should also be careful not to overwrite an existing file. I realise this is getting rather complicated now, sorry!